Source/Csla/Configuration/Fluent/DataPortalServerOptions.cs

```diff
@@ -40,7 +39,35 @@ public class DataPortalServerOptions
     /// that should be executed by the server-side data portal.
     /// injection.
     /// </summary>
-    public List<IInterceptDataPortal> InterceptorProviders { get; } = new List<IInterceptDataPortal>();
+    internal List<Type> InterceptorProviders { get; } = new List<Type>();
```

Comment:
@TheCakeMonster I think you'll like this approach for registering interceptor types better - let me know what you think.

Reply:
Yes, that looks good. Coupled with a generic method or two, plus the registration of the type against the interface, that should do us nicely :-)